File: src/tracker/config_parser.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "config_parser.h"

char* extract_tracker_address(const char* filename) {
    FILE* file = fopen(filename, "r");
    if (file == NULL) {
        perror("Erreur lors de l'ouverture du fichier");
        exit(EXIT_FAILURE);
    }

    char* tracker_address = NULL;
    char line[100];

    while (fgets(line, sizeof(line), file)) {
        if (strstr(line, "tracker-address") != NULL) {
            char* value = strchr(line, '=') + 2; // Pointer vers la valeur après le signe '='
            tracker_address = strdup(value); // Copier la valeur dans une nouvelle chaîne
            // Supprimer le saut de ligne à la fin si présent
            char* newline = strchr(tracker_address, '\n');
            if (newline != NULL) {
                *newline = '\0';
            }
            break;
        }
    }

    fclose(file);
    return tracker_address;
}

int extract_tracker_port(const char* filename) {
    FILE* file = fopen(filename, "r");
    if (file == NULL) {
        perror("Erreur lors de l'ouverture du fichier");
        exit(EXIT_FAILURE);
    }

    int tracker_port = -1;
    char line[100];

    while (fgets(line, sizeof(line), file)) {
        if (strstr(line, "tracker-port") != NULL) {
            char* value = strchr(line, '=') + 2; // Pointer vers la valeur après le signe '='
            tracker_port = atoi(value); // Convertir la valeur en entier
            break;
        }
    }

    fclose(file);
    return tracker_port;
}

File: src/pair/ConfigParser.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class ConfigParser {

    public static String extractTrackerAddress(String filename) {
        String trackerAddress = null;
        try (BufferedReader br = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = br.readLine()) != null) {
                if (line.startsWith("tracker-address")) {
                    trackerAddress = line.split("=")[1].trim();
                    break;
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return trackerAddress;
    }

    public static int extractTrackerPort(String filename) {
        int trackerPort = -1;
        try (BufferedReader br = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = br.readLine()) != null) {
                if (line.startsWith("tracker-port")) {
                    trackerPort = Integer.parseInt(line.split("=")[1].trim());
                    break;
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return trackerPort;
    }
}